### ISSUE TITLE
FR: Command palette commandline mode should store history in ApplicationState

### ISSUE THREAD

**Issue by DHowett** [microsoft/terminal#11026]
It would be very cool if I could see the 10 most recently-used (or 10 most-used!) commandlines in my command palette history. Since they're not created from the settings file, they're new user-origin content that we could save off somewhere for convenience.

**Comment by zadjii-msft** [comment]
This is a good idea

**Comment by Don-Vito** [comment]
@DHowett - wasn't aware that the AppState is merged. I will try to work on it.

**Comment by DHowett** [comment]
Yeah! It's pretty straightforward to add stuff to it, and @lhecker is on the hook for answering any questions :wink:

Thanks! Want me to assign you, or is this an advisory request?

**Comment by Don-Vito** [comment]
Actually I already have a draft :blush: Surely needs some cleanup...
@lhecker - I would love if you could take a look if I am using the state correctly

**Comment by zadjii-msft** [comment]
_the issue's been open for 6 hours and he's already got a PR_


SO glad to have you back 😝
